packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -267,91 +332,95 @@ contract PluginSetupProcessor is DaoAuthorizable {
-    /// @param _dao The address of the installing DAO.
-    /// @param _updateParams The parameters of the update.
-    /// @param _currentHelpers The address array of all current helpers (contracts or EOAs) associated with the plugin that is prepared to be updated.
-    /// @param _data The `bytes` encoded data containing the input parameters for the update as specified in the `prepareUpdateDataABI()` function in the `newPluginSetup` setup contract inside `_updateParams`.
-    /// @return permissions The list of multi-targeted permission operations to be applied to the updating DAO.
-    /// @return initData The initialization data to be passed to upgradeable contracts when the update is applied
-    /// @dev The list of `_currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which the update is prepared for.
-    function prepareUpdate(
-        address _dao,
-        PluginUpdateParams calldata _updateParams,
-        address[] calldata _currentHelpers,
-        bytes memory _data
-    ) external returns (PermissionLib.ItemMultiTarget[] memory, bytes memory) {
+    function prepareUpdate(address _dao, PrepareUpdateParams calldata params)
+        external
+        returns (PermissionLib.ItemMultiTarget[] memory, bytes memory)
+    {
+        if (
+            params.currentVersionTag.release != params.newVersionTag.release ||
+            params.currentVersionTag.build <= params.newVersionTag.build
+        ) {
+            // revert release id must be the same + new build id must be bigger.
+        }
+
         // Check that plugin is `PluginUUPSUpgradable`.
-        if (!_updateParams.plugin.supportsInterface(type(IPlugin).interfaceId)) {
-            revert IPluginNotSupported({plugin: _updateParams.plugin});
+        if (!params.plugin.supportsInterface(type(IPlugin).interfaceId)) {
+            revert IPluginNotSupported({plugin: params.plugin});
         }
-        if (IPlugin(_updateParams.plugin).pluginType() != IPlugin.PluginType.UUPS) {
-            revert PluginNonupgradeable({plugin: _updateParams.plugin});
+        if (IPlugin(params.plugin).pluginType() != IPlugin.PluginType.UUPS) {
+            revert PluginNonupgradeable({plugin: params.plugin});
         }
 
         // Implicitly confirms plugin setups are valid. // TODO Revisit this as this might not be true.
         // ensure repo for plugin manager exists
-        if (!repoRegistry.entries(address(_updateParams.pluginSetupRepo))) {
+        if (!repoRegistry.entries(address(params.pluginSetupRepo))) {
             revert PluginRepoNonexistent();
         }
 
         // Check if plugin is applied
-        if (!isInstallationApplied[_getAppliedId(_dao, _updateParams.plugin)]) {
+        if (!isInstallationApplied[_getAppliedId(_dao, params.plugin)]) {
             revert SetupNotApplied();
         }
```

Comment:
I'm not sure I follow the logic here (though it's not new). Preparing an update needs an installation to be applied. Is this relevant at this point?